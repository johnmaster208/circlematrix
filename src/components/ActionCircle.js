import React from 'react'
import PropTypes from 'prop-types'
import { Circle } from 'react-shapes'


const ActionCircle = (props) => {
  const { item, onClick, isOccupied } = props;
  return(
      <a id={item.key}
        onClick={onClick.bind(null, item)}
        >
        <Circle
          r={50}
          fill={item.fillColor}
          stroke={{color: "transparent"}}
          strokeWidth={3}
        />
      </a>
    );
};

ActionCircle.propTypes = {
  item: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired
  //onEmit: PropTypes.func.isRequired
}

export default ActionCircle
