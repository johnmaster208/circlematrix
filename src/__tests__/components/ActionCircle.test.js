import React from 'react';
import ActionCircle from '../../components/ActionCircle';
import { shallow } from 'enzyme';

describe('ActionCircle component', () => {

    const item = {
        key: 0,
        fillColor: { color: "#fff"}
    }
    const onClick = jest.fn();
    const isOccupied = false;


    it('Renders without crashing...', () => {
        const wrapper = shallow(
            <ActionCircle 
                item={item} 
                onClick={onClick} 
                isOccupied={isOccupied} />
        );
        expect(wrapper).toMatchSnapshot();
    });

});