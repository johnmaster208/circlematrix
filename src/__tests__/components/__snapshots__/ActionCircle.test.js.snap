// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionCircle component Renders without crashing... 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActionCircle
    isOccupied={false}
    item={
        Object {
            "fillColor": Object {
              "color": "#fff",
            },
            "key": 0,
          }
    }
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Circle
        className="animated pulse"
        fill={
                Object {
                        "color": "#fff",
                      }
        }
        r={50}
        stroke={
                Object {
                        "color": "transparent",
                      }
        }
        strokeWidth={3}
/>,
      "id": 0,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "animated pulse",
        "fill": Object {
          "color": "#fff",
        },
        "r": 50,
        "stroke": Object {
          "color": "transparent",
        },
        "strokeWidth": 3,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Circle
          className="animated pulse"
          fill={
                    Object {
                              "color": "#fff",
                            }
          }
          r={50}
          stroke={
                    Object {
                              "color": "transparent",
                            }
          }
          strokeWidth={3}
/>,
        "id": 0,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "animated pulse",
          "fill": Object {
            "color": "#fff",
          },
          "r": 50,
          "stroke": Object {
            "color": "transparent",
          },
          "strokeWidth": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
